core/src/main/scala/kafka/server/ReplicaFetcherThread.scala

```diff
@@ -135,6 +151,13 @@ class ReplicaFetcherThread(name: String,
     logAppendInfo
   }
 
+  private def completeDelayedFetchRequests(): Unit = {
+    if (partitionsWithNewHighWatermark.nonEmpty) {
+      replicaMgr.completeDelayedFetchRequests(partitionsWithNewHighWatermark.toSeq)
```

Comment:
I think that the failed test with Java 1.8 and Scala 2.12 is due to this line. In Scala 2.13, `toSeq` creates an immutable copy where it does not in 2.12. Hence in 2.12, the collection is cleared at the next line before the mock is verified. We should force the copy here (and add a comment).

Reply:
i'm concerned that copying the buffer will affect performance. Should we use `toIndexedSeq`?

Reply:
Yeah, I agree with you. Perhaps, we could actually change the test to cope with this instead of changing the main code. A potential solution is discussed here: https://stackoverflow.com/questions/50025561/matching-mutable-object-without-argumentcaptor. What do you think?